### ISSUE TITLE
Add Open Graph meta tags

### ISSUE THREAD

**Issue by mmmavis** [mozilla/teach.mozilla.org#798]
This can help with SEO and sharing on social media

Resources
- [The Open Graph protocol](http://ogp.me/) (this seems to be down atm)
- [How to use Open Graph Protocol](http://www.optimizesmart.com/how-to-use-open-graph-protocol/)

```
<meta property=”og:title” content=""/>
<meta property=”og:type” content=""/>
<meta property=”og:url” content=""/>
<meta property=”og:image” content=""/>
<meta property=”fb:admins” content=""/>
<meta property=”og:site_name” content=""/>
<meta property=”og:description” content=""/>
```

We'll need some copy help and image to use as site thumbnail.

**Comment by mmmavis** [comment]
I'm not sure if all the tags above are needed, e.g., `fb:admins`.  Will need to look into the docs more.

**Comment by toolness** [comment]
Good idea! For the per-page stuff, we can use a similar pattern that we're using for the [`<title>` tag content](https://github.com/mozilla/teach.webmaker.org/pull/779).

It might be useful to add the standard `<meta>` tags for `author` and `description` too?

Also, it's possible that `gulp-sitemap` will automatically see this info and include it when creating our `sitemap.xml`, but we should double-check to make sure it does.

**Comment by toolness** [comment]
I noticed that there's a cool tool at https://developers.facebook.com/tools/debug/ that lets you preview what Facebook infers a lot of the open graph tags to be. For instance, for http://teach.mozilla.org/, it shows this:

![2015-05-14_7-38-59](https://cloud.githubusercontent.com/assets/124687/7631150/5dea494a-fa0c-11e4-9493-e04e724aa6ac.jpg)

Most of this data actually seems fairly sensible, but the description is way off in particular--it looks like it's using our footer text!

**Comment by toolness** [comment]
Hmm, because this pokes at the generation of the index HTML files, it's probably a good idea to wait for #839 to land before working on this.

Also, I'm curious if facebook automatically looks at `<meta name="description">` to fill in its `og:description`. If it does, I'd prefer to use that, since it's more standards-compliant and will therefore likely work better across a wider variety of services (for instance, I think Google will use it to describe the link in search results).

**Comment by mmmavis** [comment]
> Also, I'm curious if facebook automatically looks at `<meta name="description">` to fill in its `og:description`. If it does, I'd prefer to use that, since it's more standards-compliant and will therefore likely work better across a wider variety of services (for instance, I think Google will use it to describe the link in search results).

I've done this for a few other sites before but I always included *BOTH* the standard `<meta> tags` and the Open Graph meta tags.  If I remember it correctly lots of social media sites pull info from Open Graph meta tags.  As an experiment we can try leaving Open Graph tags out, check out staging site, and see if Facebook grabs info from `<meta>` as fallback?

**Comment by mmmavis** [comment]
@hannahkane @ldecoursy @sabrinang :hamster: 

Please provide the following
- Site title (Mozilla Learning okay?)
- Site description (maybe we can grab copy from the new Homepage?)
- thumbnail to show when sharing on social media

thanks!  :grin:

**Comment by mmmavis** [comment]
https://developers.facebook.com/docs/sharing/webmasters#markup
> ... it's important that you mark up your website with Open Graph tags to take control over how your content appears on Facebook. Without these tags, the Facebook Crawler will use internal heuristics to make a best guess about the title, description, and preview image for your content

**Comment by toolness** [comment]
Cool sounds good! And duplicating the content across both meta and OG tags sounds good to me (e.g. using the same description text for `<meta>` description as well as OG description).

Should the teach team also give us succinct descriptions for each page too? According to [How to Write an Effective Meta Description ](http://blog.hubspot.com/marketing/how-to-write-meta-description-ht) they should be under 155 characters.

**Comment by mmmavis** [comment]
Coming up with per-page description might take a while.  As an interim solution could we just come up with a **site description** that we can apply on all pages?  That way at least Facebook(and possible other social media) won't be pulling text from social media [from the footer](https://cloud.githubusercontent.com/assets/124687/7631150/5dea494a-fa0c-11e4-9493-e04e724aa6ac.jpg) anymore... :neutral_face:

**Comment by ldecoursy** [comment]
How's this?

* Site title: Mozilla Learning Network (or Mozilla Learning if needs to be shorter)
* Site description: We want more people to see themselves as citizens of the web. The Mozilla Learning Network offers programs and a global community dedicated to helping people learn the most important skills of our age: the ability to read, write and participate in the digital world. 
* Thumbnail to show when sharing on social media - is there a way for it to pull options from images from the page the person is sharing? Otherwise I might suggest a portion of the main image on the Teaching Activities page https://teach.mozilla.org/activities/

Here's an alternative for a shorter site description if you need: The Mozilla Learning Network offers programs and a global community dedicated to teaching others how to read, write and participate in the digital world.

**Comment by sabrinang** [comment]
I have some thumbnail options:
Type thumbnail
![teach_typographic_thumbnail](https://cloud.githubusercontent.com/assets/2492510/7776830/feead302-0072-11e5-8007-02e3e69ce695.png)
Screen thumbnail
![teach_screen_thumbnail](https://cloud.githubusercontent.com/assets/2492510/7776832/012f615a-0073-11e5-978d-1bc2ff2600f0.png)
You can find both here: https://drive.google.com/open?id=0B4Q8pzCpDS_efkdTSlp0ajd3WllKYy1FczlxYWFoNldkek1iYVJCa0IyOWJ6eGpBWEFGNUE&authuser=0

I'd suggest the type option so it won't be dated if we update the homepage

**Comment by hannahkane** [comment]
Priority for this heartbeat is just to update the description with Lainie's copy in the ticket above. I'll file follow-up tickets for the other pieces.

**Comment by toolness** [comment]
Ok, to make this more clear, I've filed #951 to address just the og:description tag (super easy). We can address the other open graph tags later.

**Comment by mmmavis** [comment]
Since we are adding [social share buttons](https://github.com/mozilla/teach.webmaker.org/issues/1015) to the site, we should probably work on this soon.

This is the [scraped result](https://developers.facebook.com/tools/debug/og/object?q=https%3A%2F%2Fteach.mozilla.org) by using Facebook's Open Graph Object Debugger

**Comment by mmmavis** [comment]
(although it's kinda late...) I'm adding this to the current milestone (aka "Mix Tape")

**Comment by toolness** [comment]
D'oh! The picture of the gathering that was shown at https://github.com/mozilla/teach.webmaker.org/issues/798#issuecomment-102010004 isn't the default image anymore, because it's not on our front page... Maurya is cool and all but not sure if all facebook shares of our teach site should be her headshot.

For the pages that have hero unit images, I wonder if it'd be useful to set the `og:image` to the non-retina version of the hero unit image... That will require some funky coding to keep us DRY, or we could just hard-code a single image for all pages like we currently do with `og:description`.

Another alternative may be to actually use phantomjs or blitline to take a thumbnail of the page, similar to sabrina's second image in https://github.com/mozilla/teach.webmaker.org/issues/798#issuecomment-104733419, every time we deploy, though this would also take some coding work.

**Comment by ldecoursy** [comment]
Nice catch! :)

**Comment by hannahkane** [comment]
blocked on copy

**Comment by mmmavis** [comment]
Let's close this. We'll file a new ticket if what we have now on the production site is insufficient. :grin:
